src/Lucene.Net/Util/UnmanagedStringArray.cs

```diff
@@ -155,11 +199,12 @@ public int CompareTo(object other)
         private List<Segment> _segments = new List<Segment>();
 
         public int Length => _index;
-        public int _index = 1;
+        public int _index;
 
-        public UnmanagedStringArray(int size)
+        public UnmanagedStringArray(int size, int startIndex)
```

Comment:
Why can we now define `startIndex` from outside?

Reply:
This prepares for the unmanaged string usage in the Term Cache (https://github.com/ravendb/lucenenet/pull/70).

The starting index is set to 1 because it aligns with how it was defined in StringIndexCache. While I would prefer all indices to start at 0, changing this would require significantly more modifications.

Reference for context:
https://github.com/ravendb/lucenenet/blob/93c25fdc88afff8d5b51de8cc41b36296bb0b6e1/src/Lucene.Net/Search/FieldCacheImpl.cs#L801

Reply:
OK, thanks